proto/messages.go

```diff
@@ -476,10 +476,7 @@ func readRecord(r io.Reader) (*Record, error) {
 	rec.Key = dec.DecodeVarBytes()
 	rec.Value = dec.DecodeVarBytes()
 
-	len, err := dec.DecodeArrayLen()
-	if err != nil {
-		return nil, err
-	}
+	len := dec.DecodeVarInt()
```

Comment:
While we're in here, can we rename this?  `len` shadows the built-in function.

Reply:
Done.